Add only school users to game

riaJha97 opened the pull request:
Do not merge just yet.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736)
<!-- Reviewable:end -->

riaJha97 commented:
Review status: 0 of 12 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

*[a discussion](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LE9mv-a-UXpXae_gXJ8:-LE9mv-a-UXpXae_gXJ9:bwkwppy) (no related file):*
Opened up a pull request just for code review. I still can't fix the build

---


*Comments from [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736)*
<!-- Sent from Reviewable.io -->

CelineBoudier commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[.travis.yml, line 43 at r8](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LEAUVnZDqptN65EFWJs:-LEAUVnZDqptN65EFWJt:b-5o89si) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/bab13171bc69c75064c571cccd418622839c9aca/.travis.yml#L43)):*
> ```YAML
>   - pip install -U setuptools
>   - pip install -e . --process-dependency-links
>   - pip install -e .
> ```

do we still need this line as there's the other one above? or am i missing something?

---

*[setup.py, line 41 at r8](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LEAUtRZA2yS6tLIDT_6:-LEAUtRZA2yS6tLIDT_7:bb7r2f0) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/bab13171bc69c75064c571cccd418622839c9aca/setup.py#L41)):*
> ```Python
>           'sqlparse',
>           'libsass',
>           'django-forms-bootstrap'
> ```

why?

---

*[example_project/example_project/settings.py, line 71 at r8](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LEAUvnH2f-x6YLrc7kZ:-LEAUvnH2f-x6YLrc7k_:b-7jl26k) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/bab13171bc69c75064c571cccd418622839c9aca/example_project/example_project/settings.py#L71)):*
> ```Python
>     'portal',
>     'captcha',
>     'django_forms_bootstrap',
> ```

same?

---

*[portal/aimmo_game.py, line 58 at r8](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LEAVfBu8UAdIOH2LcbS:-LEAVfBu8UAdIOH2LcbT:b-du4zs7) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/bab13171bc69c75064c571cccd418622839c9aca/portal/aimmo_game.py#L58)):*
> ```Python
>     users = []
>     for player in players:
>         users.append(player.user.user)
> ```

you can also use a list comprehension! users = [player.user.user for player in players]

---

*[portal/models.py, line 78 at r8](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LEAVv06Cy2TNXypa1hJ:-LEAVv06Cy2TNXypa1hK:b-7zfwv0) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/bab13171bc69c75064c571cccd418622839c9aca/portal/models.py#L78)):*
> ```Python
>     longitude = models.CharField(max_length=20)
>     country = CountryField(blank_label='(select country)')
>     # TODO: Change this back to False
> ```

i'd rather put an issue than a todo (easier to remember)

---

*[portal/models.py, line 206 at r8](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LEAXmHHDp9QBZnkQwZA:-LEAXmHHDp9QBZnkQwZB:bhzjwkr) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/bab13171bc69c75064c571cccd418622839c9aca/portal/models.py#L206)):*
> ```Python
> 
>         user_profile = UserProfile.objects.create(user=user)
>         s = Student.objects.create(class_field=klass, user=user_profile, new_user=user)
> ```

it used to return something, not anymore. Will nothing break?

---

*[portal/migrations/0056_auto_20180530_1104.py, line 18 at r8](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LEAXzIpAWj_S7Sfo-hs:-LEAXzIpAWj_S7Sfo-ht:bj2kalv) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/bab13171bc69c75064c571cccd418622839c9aca/portal/migrations/0056_auto_20180530_1104.py#L18)):*
> ```Python
>             name='eligible_for_testing',
>             field=models.BooleanField(default=True),
>         ),
> ```

can you please rename the migration file to describe what it does? :)

---


*Comments from [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736)*
<!-- Sent from Reviewable.io -->

riaJha97 commented:
Review status: 8 of 11 files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---

*[.travis.yml, line 43 at r8](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LEAUVnZDqptN65EFWJs:-LEJEGGdBnTe-Wm7YmHI:boylvwd) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/bab13171bc69c75064c571cccd418622839c9aca/.travis.yml#L43)):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

do we still need this line as there's the other one above? or am i missing something?
</blockquote></details>

This was just something I was testing because the build kept failing on Travis. I was using this command locally to use the correct version of AI:MMO. I will have to remove this command when the other pull request I made on aimmo gets merged in.

---

*[setup.py, line 41 at r8](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LEAUtRZA2yS6tLIDT_6:-LEJEktGC727T0ECvN2a:b2s2gih) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/bab13171bc69c75064c571cccd418622839c9aca/setup.py#L41)):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

why?
</blockquote></details>

Running portal locally wouldn't allow me to access aimmo because there was a TemplateSyntaxError on {% load bootstrap tags %} that is fixed by adding the line above

---

*[example_project/example_project/settings.py, line 71 at r8](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LEAUvnH2f-x6YLrc7kZ:-LEJFObP6dZZhpyhWKlb:bnj7hl0) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/bab13171bc69c75064c571cccd418622839c9aca/example_project/example_project/settings.py#L71)):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

same?
</blockquote></details>

same as above

---

*[portal/aimmo_game.py, line 58 at r8](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LEAVfBu8UAdIOH2LcbS:-LEJG_dR2aQJkNMBGnw3:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/bab13171bc69c75064c571cccd418622839c9aca/portal/aimmo_game.py#L58)):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

you can also use a list comprehension! users = [player.user.user for player in players]
</blockquote></details>

Done.

---

*[portal/models.py, line 78 at r8](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LEAVv06Cy2TNXypa1hJ:-LEJH3rJ3RmTRB3scSnH:b-lmx0kw) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/bab13171bc69c75064c571cccd418622839c9aca/portal/models.py#L78)):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

i'd rather put an issue than a todo (easier to remember)
</blockquote></details>

This was just so I could test on dev but yeah I'll do that

---

*[portal/models.py, line 206 at r8](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LEAXmHHDp9QBZnkQwZA:-LEJMU0r7tJJIy8wmH5M:b-sm90ix) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/bab13171bc69c75064c571cccd418622839c9aca/portal/models.py#L206)):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

it used to return something, not anymore. Will nothing break?
</blockquote></details>

I removed this bit , it was unnecessary

---

*[portal/migrations/0056_auto_20180530_1104.py, line 18 at r8](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LEAXzIpAWj_S7Sfo-hs:-LEJMwsjDiy4tXfeXk_F:b7qjrh5) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/bab13171bc69c75064c571cccd418622839c9aca/portal/migrations/0056_auto_20180530_1104.py#L18)):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

can you please rename the migration file to describe what it does? :)
</blockquote></details>

I've renamed the file but I'll be deleting this before merging because this migration was only for testing on dev

---


*Comments from [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736)*
<!-- Sent from Reviewable.io -->

CelineBoudier commented:
Reviewed 3 of 4 files at r10.
Review status: all files reviewed, 2 unresolved discussions (waiting on @riaJha97)

---

*[portal/aimmo_game.py, line 42 at r9](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LEz1X0YBL4q525C0pXe:-LEz1X0YBL4q525C0pXf:breh8mi) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/1fec7e0d3b011c792b3f45d953be8c581cf4b1fd/portal/aimmo_game.py#L42)):*
> ```Python
> 
> def get_users_for_new_game(request):
>     print "Inside portal's get users function"
> ```

will have to remove this ;)

---


*Comments from [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736)*
<!-- Sent from Reviewable.io -->

riaJha97 commented:
Review status: all files reviewed, 2 unresolved discussions (waiting on @riaJha97)

---

*[portal/aimmo_game.py, line 42 at r9](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LEz1X0YBL4q525C0pXe:-LF1difHDAK_bnatjEr5:b-4qz2za) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/1fec7e0d3b011c792b3f45d953be8c581cf4b1fd/portal/aimmo_game.py#L42)):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

will have to remove this ;)
</blockquote></details>

I've removed the print statement, I'm not sure why it's still showing up here :/

---


*Comments from [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736)*
<!-- Sent from Reviewable.io -->

OlafSzmidt commented:
Reviewed 5 of 11 files at r7, 2 of 3 files at r9, 2 of 4 files at r10.
Review status: all files reviewed, 3 unresolved discussions (waiting on @riaJha97)

---

*[a discussion](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LE9mv-a-UXpXae_gXJ8:-LF1r7daEvpDi-Zl3ZWh:br8b0g8) (no related file):*
<details><summary><i>Previously, riaJha97 wrote…</i></summary><blockquote>

Opened up a pull request just for code review. I still can't fix the build
</blockquote></details>

Is everything fixed now with the build?

---

*[setup.py, line 43 at r10](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LF1rDCf4xVVc8BO9Iod:-LF1rDCf4xVVc8BO9Ioe:bldwofb) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/f8a4407b6675911f615044fda05e3da6dc83fc2a/setup.py#L43)):*
> ```
> #egg=aimmo-0.1.1a0.post0.dev56
> ```
This will need updating every time right? How can we automate this?

---

*[portal/templatetags/app_tags.py, line 86 at r10](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LF1reag7w6Hocmy76AM:-LF1reag7w6Hocmy76AN:b-fusrzv) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/f8a4407b6675911f615044fda05e3da6dc83fc2a/portal/templatetags/app_tags.py#L86)):*
> ```
> is_developer(u)
> ```
How is `userprofile.developer` assigned to a user again? Is it manually through the db?

---


*Comments from [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736)*
<!-- Sent from Reviewable.io -->

CelineBoudier commented:
Reviewed 1 of 3 files at r11.
Review status: 7 of 8 files reviewed, 4 unresolved discussions (waiting on @CelineBoudier, @OlafSzmidt, and @riaJha97)

---

*[portal/models.py, line 78 at r11](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LF2dJcJ2dXkQGi2hQLg:-LF2dJcK2hsfSS58gind:b-ifj2ej) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/e8ef6ee7222b2cc1c3e07224aea61f07a9ccf6e6/portal/models.py#L78)):*
> ```Python
>     longitude = models.CharField(max_length=20)
>     country = CountryField(blank_label='(select country)')
>     eligible_for_testing = models.BooleanField(default=False)
> ```

so we don't need the migration anymore or?

---


*Comments from [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736)*
<!-- Sent from Reviewable.io -->

riaJha97 commented:
Review status: 7 of 8 files reviewed, 4 unresolved discussions (waiting on @CelineBoudier, @OlafSzmidt, and @riaJha97)

---

*[portal/models.py, line 78 at r11](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LF2dJcJ2dXkQGi2hQLg:-LF2gMNT3VOm9Cmr1P9z:b-82s9mm) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/e8ef6ee7222b2cc1c3e07224aea61f07a9ccf6e6/portal/models.py#L78)):*
<details><summary><i>Previously, CelineBoudier (Celine Boudier) wrote…</i></summary><blockquote>

so we don't need the migration anymore or?
</blockquote></details>

The migration I had initially in the pull request had all schools eligible for using aimmo, which I was just using for testing (assuming that selected schools will be manually made eligible for testing in the db). The migration file 0055 (already on master) has eligible_for_testing set to false by default so this change should be fine

---


*Comments from [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736)*
<!-- Sent from Reviewable.io -->

OlafSzmidt commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 3 files at r11.
Review status: all files reviewed, 2 unresolved discussions (waiting on @riaJha97 and @CelineBoudier)

---



*Comments from [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-:-LFHb91j7OvqcYliyVNk:bnfp4nl)*
<!-- Sent from Reviewable.io -->

riaJha97 commented:
The pull request in aimmo must be merged before this

---

Review status: all files reviewed, 1 unresolved discussion (waiting on @CelineBoudier)

---

*[portal/models.py, line 78 at r11](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-LF2dJcJ2dXkQGi2hQLg:-LFIaB5G3lhz2sjzzwo1:b-896fix) ([raw file](https://github.com/ocadotechnology/codeforlife-portal/blob/e8ef6ee7222b2cc1c3e07224aea61f07a9ccf6e6/portal/models.py#L78)):*
<details><summary><i>Previously, riaJha97 wrote…</i></summary><blockquote>

The migration I had initially in the pull request had all schools eligible for using aimmo, which I was just using for testing (assuming that selected schools will be manually made eligible for testing in the db). The migration file 0055 (already on master) has eligible_for_testing set to false by default so this change should be fine
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-:-LFIaG8_6xSn5rH4iLkr:bd0n1jh)*
<!-- Sent from Reviewable.io -->

CelineBoudier commented:
Reviewed 1 of 3 files at r11.
Review status: :shipit: complete! all files reviewed, all discussions resolved

---



*Comments from [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736)*
<!-- Sent from Reviewable.io -->

CelineBoudier commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: :shipit: complete! all files reviewed, all discussions resolved

---



*Comments from [Reviewable](https://reviewable.io/reviews/ocadotechnology/codeforlife-portal/736#-:-LFN827c4HC6y0HZI2gH:bnfp4nl)*
<!-- Sent from Reviewable.io -->
